src/util/NamedColor.h

```diff
@@ -0,0 +1,112 @@
+/**
+ * Xournal++
+ *
+ * Part of the customizable toolbars
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ * @brief Struct for NamedColor
+ */
+
+#pragma once
+#include <string>
+
+#include "util/Color.h"
+
+struct Palette;
+/**
+ * @brief NamedColor is an object for a color with name and an optional index which refers to the index of the color
+ * inside a given palette.
+ * If the NamedColor does not come from a Color palette isColorPalette is set to false and the palettIndex is 0.
+ *
+ */
+struct NamedColor {
+    /**
+     * @brief Default Construct a new Named Color instance
+     * This is only needed for the input operator implementation
+     *
+     */
+    NamedColor();
+
+    /**
+     * @brief Construct a new NamedColor instance for a Color Palette
+     * The paletteIndex should be the index which the color will have in the palette.
+     * This is to be used as:
+     *
+     *     NamedColor color{namedColors.size()};
+     *     std::istringstream iss{line};
+     *     if (iss >> color) {
+     *        ...
+     *     }
+     *
+     * @param paletteIndex
+     */
+    NamedColor(const size_t& paletteIndex);
+
+    /**
+     * @brief Construct a new NamedColor instance
+     * This automatically sets the isPaletteColor to negative as NamedColors from a color palette can only be
+     * constructed using the input operator.
+     * As it is no paletteColor the Name of the Color will be set to "Custom Color" and the paletteIndex to 0.
+     *
+     * @param color color to set for the custom NamedColor instance
+     */
+    NamedColor(const Color& color);
```

Comment:
On 32 bit system, I cannot compile because size_t is uint32_t, so NamedColor::NamedColor (const Color &) cannot overload NamedColor::NamedColor (const size_t &). I can bypass the problem by replacing the type of the paletteIndex member with uint64_t, but surely there must be a better way to fix it.

Reply:
Thanks for reporting this :pray: .
I think the easiest is to use `uint16_t` instead of `size_t` for the `paletteIndex`. I guess it's rather unlikely that we'll ever have more than 65536 colors in a palette. I'm already on it and will raise a PR soon.

Reply:
@LittleHuba : Well I guess that's another reason I shouldn't use `size_t` that much anymore :sweat_smile:

Reply:
@marioalexis84 : I prepared a fix at #3616. Please check if that works for you.

Reply:
> @LittleHuba : Well I guess that's another reason I shouldn't use `size_t` that much anymore :sweat_smile: 

Yeah one of the common issues of C(++). If I remember correctly, the only thing keeping us from making official 32-bit releases is the use of size_t in various places.

Reply:
> Well I guess that's another reason I shouldn't use size_t that much anymore 😅

Don't be that harsh to yourself: Using `size_t` for indices, especially when accessing containers, is completely fine.
Also, the usage of size_t is only problematic, if it is mixed with, or assigned to, other types.

This is rather an example to use stronger types than to not use size_t.
E.g. use my Color debug struct, to enforce that really every rgba-color is an uint32_t.